build/common.sh

```diff
@@ -679,16 +678,19 @@ function kube::build::sync_to_container() {
   # directory and generated files. The '- /' filter prevents rsync
   # from trying to set the uid/gid/perms on the root of the sync tree.
   # As an exception, we need to sync generated files in staging/, because
-  # they will not be re-generated by 'make'.
+  # they will not be re-generated by 'make'. Note that the 'H' filtered files
```

Comment:
Why do we jump through this hoop?  why don't we rsync EVERYTHING?  Just treat it like a volume mount.  As more and more scripts run in the container, we'll have to sync more and more anyway...

Reply:
Mainly due to bandwidth constraints around remote builds. https://github.com/kubernetes/kubernetes/pull/46193#issuecomment-320662223 and https://github.com/kubernetes/kubernetes/issues/50272#issue-248542031 have more details. I tried to sync all of `/.git/` in [here](https://github.com/kubernetes/kubernetes/commit/f9f43143a4deb031b3565bddebc13bd0dd3e9da5#diff-0ee501c7b4b1ab61678579826c6d6534L686) but there was push back.

Reply:
ok, ignoring .git makes sense, but really it only hurts once, right?  Everything else is noise by comparison.  Why not sync everything but .git ?

Reply:
So if we keep the filter for `/.git/`, then we're left with:
- `/.make/` - make-related metadata that is gitignored
- `/_tmp/` - precommit temporary directory created by ./hack/verify-generated-docs.sh and ./hack/lib/util.sh. also gitignored.
- `/_output/` - Designed to be a one way sync out of the container, rather than something you sync in
- `zz_generated.*` - Generated files. Could allow these to sync, but likely of minimal use.
- `generated.proto` - Generated files. Could allow these to sync, but likely of minimal use.

The generated files, I'm okay with syncing in.. but the gitignored directories are probably best to not sync in, as it makes the container environment more hermetic.